crypto/evp/kdf_lib.c

```diff
@@ -132,12 +132,19 @@ int EVP_KDF_CTX_ctrl_str(EVP_KDF_CTX *ctx,
     return ret;
 }
 
+size_t EVP_KDF_CTX_output_size(EVP_KDF_CTX *ctx)
+{
+    if (ctx == NULL || ctx->kmeth->output_size == NULL)
```

Comment:
Wouldn't it be better to return an error if ctx is NULL ?
Given 0 is used to signifiy variable length, the return value would be wrong as a generic return value ...

Reply:
Yes, 0 can mean either variable length or an error.

What value would you suggest returning in the case of an error, then? `SIZE_MAX`?
We could also change the return type to either `int` or `ossl_ssize_t` and return -1.

Alternatively, we could just say that attempting to get the output size for a variable-output-size algorithm is an error and therefore 0 always represents an error.

Thoughts?

Currently only HKDF in EVP_KDF_HKDF_MODE_EXTRACT_ONLY will produce a fixed output size.
An error can occur if either `ctx` is `NULL` or the `MD` has not yet been set.

Reply:
I do not know what's the best way to proceed, I guess "just say that attempting to get the output size for a variable-output-size algorithm is an error" may be a good way to put it ...

Reply:
We could also remove the `NULL` check for `ctx` and state that returning 0 for `MD` not being set just indicates that the output size is not yet known. I.e. errors don't actually occur.

Another alternative... return SIZE_MAX for variable output size and 0 for errors.

Reply:
After giving it some thought I ended up using SIZE_MAX to indicate variable output size and 0 for errors.